extensions/api/auth-tokenbased/README.md

```diff
@@ -0,0 +1,19 @@
+# Token Based Authentication Service
+
+The token based authentication service extension is used to secure connector APIs. These APIs are not protected by the `AuthenticationService` by default. To find out how a specific API is protected please consolidate its documentation.
+
+APIs, protected by this extension, require a client to authenticate by adding a authentication key to the request header.
+
+Authentication Header Example:
+```
+curl <url> --header "X-API-Key: <key>"
+```
+
+## Configuration
+
+| Key | Description | Required |
+|:---|:---|:---|
+| edc.api.auth.key | API Key Header Value | false |
+| edc.api.auth.key.alias | Secret name of the API Key Header Value, stored in the vault | false |
+
+If no authentication key is defined, a random value is generated and printed out into the logs.
```

Comment:
I think it should be made clear
- that the API key can be stored in the `Vault` _and_ in the configuration
- what the order of precedence is, i.e. what happens if there's a key in the vault AND the config, etc.